cbor_event/src/de.rs

```diff
@@ -546,6 +560,11 @@ impl<'a> RawCbor<'a> {
         self.special()?.unwrap_bool()
     }
 
+    pub fn eof(&mut self) -> Result<()> {
+        if !self.0.is_empty() { return Err(Error::TrailingData) }
+        Ok(())
+    }
```

Comment:
I am not sure this function is needed. There is already an implementation for `Deref` which allows to access all the API of `[u8]`. So the user can already do:

```rust
let raw = RawCbor::from(&[]);

assert!(raw.is_empty()); // instead of raw.eof()
```

if you move `decode_complete` in this module, you could keep the useful error though:

```rust
impl<'a> RawCbor<'a> {
    /// just like `deserialize` but expects to consume all the data in the RawCbor
    /// (i.e. no trailing data).
    fn deserialize_complete<T>(&mut self) -> Result<T>
        where T: Deserialize
    {
        let v = self.deserialize()?;
        if ! self.is_empty() {
            Err(Error::TrailingData)
        } else { Ok(v) }
    }
}
```